samples/nrf9160/nrf_cloud_coap_client/src/main.c

```diff
@@ -0,0 +1,807 @@
+		if (cur_rrc_mode == LTE_LC_RRC_MODE_IDLE) {
+			LOG_DBG("RRC mode: idle");
+		} else {
+			LOG_DBG("RRC mode: connected");
+		}
+		if (request_cells && (cur_rrc_mode == LTE_LC_RRC_MODE_IDLE)) {
+			get_cell_info();
+		}
+		break;
+	case LTE_LC_EVT_NEIGHBOR_CELL_MEAS:
+		if ((search_type < LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_DEFAULT) &&
+		    (evt->cells_info.current_cell.id == LTE_LC_CELL_EUTRAN_ID_INVALID)) {
+			LOG_WRN("Current cell ID not valid in neighbor cell measurement results");
+			break;
+		}
+
+		(void)k_mutex_lock(&cell_info_mutex, K_FOREVER);
+		/* Copy current cell information. */
+		memcpy(&cell_info.current_cell,
+		       &evt->cells_info.current_cell,
+		       sizeof(cell_info.current_cell));
+
+		/* Copy neighbor cell information if present. */
+		cell_info.ncells_count = evt->cells_info.ncells_count;
+		if ((cell_info.ncells_count > 0) && (evt->cells_info.neighbor_cells)) {
+			memcpy(neighbor_cells,
+			       evt->cells_info.neighbor_cells,
+			       sizeof(neighbor_cells[0]) * cell_info.ncells_count);
+			LOG_DBG("Received measurements for %u neighbor cells",
+				cell_info.ncells_count);
+		} else {
+			LOG_DBG("No neighbor cells were measured");
+		}
+
+		/* Copy GCI cell information if present. */
+		cell_info.gci_cells_count = evt->cells_info.gci_cells_count;
+		if ((cell_info.gci_cells_count > 0) && (evt->cells_info.gci_cells)) {
+			memcpy(gci_cells,
+			       evt->cells_info.gci_cells,
+			       sizeof(gci_cells[0]) * cell_info.gci_cells_count);
+			LOG_DBG("Received measurements for %u GCI cells",
+				cell_info.gci_cells_count);
+		} else if (search_type == LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_EXTENDED_COMPLETE) {
+			LOG_DBG("No GCI cells were measured");
+		}
+
+		(void)k_mutex_unlock(&cell_info_mutex);
+		k_sem_give(&cell_info_ready_sem);
+
+		break;
+	default:
+		LOG_DBG("LTE event %d (0x%x)", evt->type, evt->type);
+		break;
+	}
+}
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+void pgps_handler(struct nrf_cloud_pgps_event *event)
+{
+	switch (event->type) {
+	case PGPS_EVT_INIT:
+		LOG_INF("PGPS_EVT_INIT");
+		break;
+	case PGPS_EVT_UNAVAILABLE:
+		LOG_INF("PGPS_EVT_UNAVAILABLE");
+		break;
+	case PGPS_EVT_LOADING:
+		LOG_INF("PGPS_EVT_LOADING");
+		break;
+	case PGPS_EVT_AVAILABLE:
+		LOG_INF("PGPS_EVT_AVAILABLE");
+		break;
+	case PGPS_EVT_READY:
+		LOG_INF("PGPS_EVT_READY");
+		pgps_active = false;
+		break;
+	case PGPS_EVT_REQUEST:
+		LOG_INF("PGPS_EVT_REQUEST");
+		pgps_active = true;
+		do_pgps(event->request);
+		break;
+	}
+}
+#endif
+
+/**@brief Configures modem to provide LTE link. Blocks until link is
+ * successfully established.
+ */
+static int modem_configure(void)
+{
+	int err;
+
+	err = nrf_modem_lib_init();
+	if (err < 0) {
+		LOG_ERR("Modem library initialization failed, error: %d", err);
+		return err;
+	} else if (err > 0) {
+		LOG_INF("Modem firmware update in progress: 0x%X", err);
+		return err;
+	}
+
+	lte_lc_register_handler(lte_handler);
+#if defined(CONFIG_LTE_LINK_CONTROL)
```

Comment:
I'm not sure I understand what your suggestion is.

Reply:
Instead of the `CONFIG_LTE_LINK_CONTROL` guard on just the section that uses `lte_lc`, you could guard the entire function for modem being enabled (since it doesn't seem to me like modem will be enabled in this sample without `lte_lc` being enabled)

It's a minor thing, though. `CONFIG_LTE_LINK_CONTROL` is arguably more specific, but I do suspect you'll eventually need to guard for modem enablement anyways

Reply:
This originally came from the sdk-nrf `coap_client` sample.

Reply:
Huh, interesting. I'm not sure why they would support builds both with and without `lte_lc`